pkg/resource/predicates.go

```diff
@@ -43,34 +41,32 @@ func NewPredicates(fn PredicateFn) predicate.Funcs {
 	}
 }
 
-// ObjectHasProvisioner returns a PredicateFn implemented by HasProvisioner.
-func ObjectHasProvisioner(c client.Client, provisioner string) PredicateFn {
+// ObjectHasClassKind returns a PredicateFn implemented by HasClassKind.
+func ObjectHasClassKind(c client.Client, cs Class) PredicateFn {
```

Comment:
Yep! I believe the effect of doing this would mean that if the class that is referenced didn't actually exist then we would fail in the reconcile loop as opposed to just not even trying to reconcile in the current implementation. Are we okay with that? I actually like it better because it will give the user an actionable error.

Reply:
(i.e. `kubectl describe` the claim will tell what is wrong as opposed to the predicate which will just prevent reconciliation.

Reply:
> Are we okay with that? I actually like it better because it will give the user an actionable error.

Agreed.

It's not ideal that the policy author won't learn that the default resource class they've set is invalid until it's propagated to a resource claim, but at least the resource claim author can see the error condition and report it to the policy author.

We could provide direct feedback to the policy author by adding controllers that watch for policies and validate them, but I don't think that needs to be in scope for our initial pass at this.